rust-bindings/src/lib.rs

```diff
@@ -36,6 +36,34 @@ pub fn validate_proof(
 mod tests {
     use super::*;
 
+    #[test]
+    fn test_proofs() {
+        /*
+         * Generated by printing the inputs to validate_proof in tests/test_python_bindings.py
+         * Specifically, the test_k_21 function was adapted to append 100 lines to the file.
+         */
+        let proofs = include_str!("../test_proofs.txt");
+
+        for line in proofs.lines() {
+            let mut parts = line.split(", ");
+            let seed = hex::decode(parts.next().unwrap()).unwrap();
+            let k = parts.next().unwrap().parse().unwrap();
+            let challenge = hex::decode(parts.next().unwrap()).unwrap();
+            let proof = hex::decode(parts.next().unwrap()).unwrap();
+            let quality = hex::decode(parts.next().unwrap()).unwrap();
+
+            let mut actual_quality = [0; 32];
+            assert!(validate_proof(
+                &seed.try_into().unwrap(),
+                k,
+                &challenge.try_into().unwrap(),
+                &proof,
+                &mut actual_quality
+            ));
+            assert_eq!(actual_quality.as_slice(), quality);
```

Comment:
it might make sense to also ensure that `validate_proof()` *fails* if you corrupt any of the inputs.